src/main/java/mate/academy/internetshop/dao/jdbcimpl/ProductJdbcImpl.java

```diff
@@ -0,0 +1,106 @@
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                element.setId(resultSet.getLong(1));
+            }
+            return element;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t add product to DB", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long element) {
+        String sql = String.format("SELECT * FROM product WHERE product_id=%d;", element);
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(sql);
+            ResultSet resultSet = statement.executeQuery();
+            Product product = null;
+            while (resultSet.next()) {
+                product = getProductResultSet(resultSet);
+            }
+            return Optional.ofNullable(product);
+        } catch (SQLException e) {
+            throw new DataProcessingException("can`t receive product from DB by id ", e);
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        List<Product> products = new ArrayList<>();
+        String sql = "SELECT * FROM product;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(sql);
+            ResultSet resultSet = statement.executeQuery();
+            Product product;
+            while (resultSet.next()) {
+                product = getProductResultSet(resultSet);
+                products.add(product);
+            }
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t receive all products from DB", e);
+        }
+    }
+
+    @Override
+    public Product update(Product element) {
+        String sql = "UPDATE product SET name=?, price=? WHERE product_id="
+                + element.getId() + ";";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(sql);
+            statement.setString(1, element.getName());
+            statement.setDouble(2, element.getPrice());
+            return get(element.getId()).get();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update product in DB", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long element) {
+        String sql = "DELETE FROM product WHERE product_id=" + element + ";";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            Statement statement = connection.createStatement();
+            return statement.executeUpdate(sql) > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete product from DB", e);
+        }
+    }
+
+    private Product getProductResultSet(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private Product getProductFromResultSet(ResultSet resultSet) throws SQLException {
```
так звучить логічніше